content/blog/2020-03-01-bookdown-learnings/index.Rmd

```diff
@@ -0,0 +1,257 @@
+
+Practically, I kept the usethis [pull request helpers](https://usethis.r-lib.org/articles/articles/pr-functions.html) article open in my browser and forced myself to use them.
+At the start of the day, I would open our bookdown RStudio project by clicking my local `ropensci-blog-guidance.Rproj` file, pull to update my local master (because Maëlle would have reviewed, edited and merged my pull requests while I slept), create a new branch with `pr_init(branch = "branchname")`, make edits, and `pr_push()` to push my local changes to GitHub master.
+usethis automatically opened a browser to the GitHub web interface prompting me to open a pull request. 
+
+#### 4. How to fill up a bunch of Rmd files to make a book
+
+To answer the question, "how do I do this thing?" I often compared the 
+GitHub file structure and contents of completed books with their corresponding live pages. This, for me, had the biggest payoff in learning how to bookdown.
+
+Things I learned:
+
+- One book chapter is made from one Rmd file.
+- Chapters are woven together in the [`_bookdown.yml`](https://github.com/ropensci-org/blog-guidance/blob/master/_bookdown.yml) file that references those Rmd files
+- Create links inside the book with `{#anchor}`. In the [`authorcontent.Rmd`](https://github.com/ropensci-org/blog-guidance/blob/master/authorcontent.Rmd) file, the heading `# Content Guidelines {#content}` means that a markdown format `[link to that chapter](#content)` anywhere in the book will link to the Content Guidelines chapter.
+- Appendices A to H (_too many; we know_) are created from a single [appendix.Rmd](https://github.com/ropensci-org/blog-guidance/blob/master/appendix.Rmd) file made up of groupings of a heading, some text, and sometimes a code chunk that points to a file, like a [template](https://github.com/ropensci-org/blog-guidance/tree/master/templates) or [checklist](https://github.com/ropensci-org/blog-guidance/tree/master/checklists) that populates that appendix.
+
+<!--html_preserve--> {{< figure src = "file-comp.png" width = "800" alt = "this is the alternative text" >}}<!--/html_preserve-->
+
+I often felt quite euphoric about the things I was able to figure out comparing GitHub file structures to their books. 
+(Consisentely, within days I would take this hard-earned knowledge for granted and feel inadequate in the face of my next technical challenge...until I felt the satisfaction of owning that next one too.)
+
+These bookdown books were helpful in exploring those comparisons:
+
+- rOpenSci Packages: Development, Maintenance, and Peer Review [GitHub](https://github.com/ropensci/dev_guide) _cf_ [Book](https://devguide.ropensci.org/)
+- R for Data Science [GitHub](https://github.com/hadley/r4ds) _cf_ [Book](https://r4ds.had.co.nz/)
+
+#### 5. How to everything
```

Comment:
Lol I want the title to remain this

Reply:
But a common theme between trouble shooting and linking issue is finding an error and fixing it. :thinking:

Reply:
I like that title - was temporary but intentional. Cause it felt like this bookdown project really was learning "how to everything" 😆